source/gameengine/GameLogic/Joystick/SCA_JoystickEvents.cpp

```diff
@@ -86,48 +70,73 @@ void SCA_Joystick::HandleEvents(void)
 		return;
 	}
 
-	int i;
-	for (i=0; i<m_joynum; i++) { /* could use JOYINDEX_MAX but no reason to */
+	for (int i = 0; i < JOYINDEX_MAX; i++) {
 		if (SCA_Joystick::m_instance[i])
 			SCA_Joystick::m_instance[i]->OnNothing(&sdl_event);
 	}
 	
 	while (SDL_PollEvent(&sdl_event)) {
-		/* Note! m_instance[sdl_event.jaxis.which]
+		/* Note! m_instance[instance]
 		 * will segfault if over JOYINDEX_MAX, not too nice but what are the chances? */
 		
 		/* Note!, with buttons, this wont care which button is pressed,
-		 * only to set 'm_istrig_button', actual pressed buttons are detected by SDL_JoystickGetButton */
+		 * only to set 'm_istrig_button', actual pressed buttons are detected by SDL_ControllerGetButton */
 		
 		/* Note!, if you manage to press and release a button within 1 logic tick
 		 * it wont work as it should */
+
+		/* Note!, we need to use SDL_JOYDEVICE ADDED to find new controllers as SDL_CONTROLLERDEVICEADDED
+		 * doesn't report about all devices connected at beginning. Additionally we capture all devices this
+		 * way and we can inform properly (with ways to solve it) if the joystick it is not a game controller */
 		
 		switch (sdl_event.type) {
-			case SDL_JOYAXISMOTION:
-				SCA_Joystick::m_instance[sdl_event.jaxis.which]->OnAxisMotion(&sdl_event);
-				break;
-			case SDL_JOYHATMOTION:
-				SCA_Joystick::m_instance[sdl_event.jhat.which]->OnHatMotion(&sdl_event);
-				break;
-			case SDL_JOYBUTTONUP:
-				SCA_Joystick::m_instance[sdl_event.jbutton.which]->OnButtonUp(&sdl_event);
+			case SDL_JOYDEVICEADDED:
+				if (sdl_event.jdevice.which < JOYINDEX_MAX) {
+					if (!SCA_Joystick::m_instance[sdl_event.jdevice.which]) {
```

Comment:
I think it would be interesting to print a message if an added joystick is added on an existing slot.